modules/Kconfig.mcux

```diff
@@ -5,7 +5,8 @@
 
 config HAS_MCUX
 	bool
-	select HAS_CMSIS_CORE
+	select HAS_CMSIS_CORE if SOC_FAMILY_KINETIS || SOC_FAMILY_IMX || \
```

Comment:
considering all other families except ADSP needs this symbol, wouldn't be simpler to do `select HAS_CMSIS_CORE if !SOC_FAMILY_NXP_ADSP`